src/main/java/dev/gigaherz/graph3/Mergeable.java

```diff
@@ -19,8 +19,7 @@ public Dummy mergeWith(Dummy other)
         }
 
         @Override
-        public Dummy copy()
-        {
+        public Dummy splitFor(int selfNodeCount, int totalNodeCount) {
```

Comment:
Hmm, I have a number of concerns with this approach:
- First, that there's no way to know how many graphs it split into, only the nodes, meaning someone couldn't choose to split evenly based on graphs instead of nodes,
- Second, that it doesn't provide a means to deal with remainders (e.g. 100 units of data between 3 graphs, 1 of them should get 34 units otherwise 1 unit would be lost), and
- Finally, to avoid unnecessary breaking changes, it would be best to keep the copy method around, with deprecation annotation.

(I'm going to sleep in a moment so any replies to this I'll read tomorrow)

Reply:
I'll re-add the copy() method, and default splitFor() to the copy() method for compatibility.

As for the graph count, we can easily add information about the number of graphs being split into.

I'm unsure how to handle the remainders however - do you have any suggestions?
Possibly implementations should be pessimistic about overflow, always tending to Math.floor for calculations, and then they can use the graph count (and graph index would need to be added) to discern whether they should allocate all the remainder to the final graph?

Reply:
I have pushed my proposal for handling these requirements, please let me know if you still reckon there's a different way to go. I'll also screenshot how I handled this in Ender IO's energy storage to demonstrate how I envision this being adequate.

<img width="909" alt="image" src="https://github.com/user-attachments/assets/83b30650-ccb2-49ce-b7d9-fc4e0106de91" />

Reply:
Sorry this week has been crazy busy (and the busy might last a few days more). Will try to look at this soonish.

Reply:
No worries at all! I hope you have a merry Christmas when it arrives :)